pkg/controller/istio/istio_controller_test.go

```diff
@@ -303,4 +309,477 @@ var _ = Describe("Istio controller tests", func() {
+			Expect(*updatedFC.Spec.IstioAmbientMode).To(Equal("Enabled"))
+			Expect(updatedFC.Annotations).To(HaveKey(istio.IstioOperatorAnnotationMode))
+			Expect(updatedFC.Annotations[istio.IstioOperatorAnnotationMode]).To(Equal("Enabled"))
+		})
+
+		It("should patch FelixConfiguration with IstioDSCPMark", func() {
+			fc := &crdv1.FelixConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+			}
+			Expect(cli.Create(ctx, fc)).NotTo(HaveOccurred())
+
+			dscpMark := numorstring.DSCPFromInt(23)
+			istioCR.Spec.DSCPMark = &dscpMark
+			Expect(cli.Update(ctx, istioCR)).NotTo(HaveOccurred())
+
+			r := &ReconcileIstio{
+				Client:   cli,
+				scheme:   scheme,
+				provider: operatorv1.ProviderNone,
+				status:   mockStatus,
+			}
+
+			_, err := r.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: "default"}})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Verify FelixConfiguration was patched with DSCP mark
+			updatedFC := &crdv1.FelixConfiguration{}
+			Expect(cli.Get(ctx, types.NamespacedName{Name: "default"}, updatedFC)).NotTo(HaveOccurred())
+			Expect(updatedFC.Spec.IstioDSCPMark).NotTo(BeNil())
+			Expect(updatedFC.Spec.IstioDSCPMark.ToUint8()).To(Equal(uint8(23)))
+			Expect(updatedFC.Annotations).To(HaveKey(istio.IstioOperatorAnnotationDSCP))
+			Expect(updatedFC.Annotations[istio.IstioOperatorAnnotationDSCP]).To(Equal("23"))
+		})
+
+		It("should handle custom DSCPMark value", func() {
+			fc := &crdv1.FelixConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+				},
+			}
+			Expect(cli.Create(ctx, fc)).NotTo(HaveOccurred())
+
+			customDSCP := numorstring.DSCPFromInt(10)
+			istioCR.Spec.DSCPMark = &customDSCP
+			Expect(cli.Update(ctx, istioCR)).NotTo(HaveOccurred())
+
+			r := &ReconcileIstio{
+				Client:   cli,
+				scheme:   scheme,
+				provider: operatorv1.ProviderNone,
+				status:   mockStatus,
+			}
+
+			_, err := r.Reconcile(ctx, reconcile.Request{NamespacedName: types.NamespacedName{Name: "default"}})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Verify FelixConfiguration has custom DSCP mark
+			updatedFC := &crdv1.FelixConfiguration{}
+			Expect(cli.Get(ctx, types.NamespacedName{Name: "default"}, updatedFC)).NotTo(HaveOccurred())
+			Expect(updatedFC.Spec.IstioDSCPMark).NotTo(BeNil())
+			Expect(updatedFC.Spec.IstioDSCPMark.ToUint8()).To(Equal(uint8(10)))
+			Expect(updatedFC.Annotations[istio.IstioOperatorAnnotationDSCP]).To(Equal("10"))
+		})
+
+		It("should detect user modification of IstioAmbientMode in FelixConfiguration", func() {
+			// Create FelixConfiguration with mismatched annotation and spec
+			userModifiedMode := "UserModified"
+			fc := &crdv1.FelixConfiguration{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "default",
+					Annotations: map[string]string{
+						istio.IstioOperatorAnnotationMode: "Enabled",
+					},
+				},
+				Spec: crdv1.FelixConfigurationSpec{
+					IstioAmbientMode: &userModifiedMode,
```

Comment:
```suggestion
					IstioAmbientMode: ptr.To("Disabled"),
```

We should use realistic values in the tests, and we can avoid the local var using the `ptr` library.